app/models/concerns/eds_links.rb

```diff
@@ -0,0 +1,73 @@
+module EdsLinks
+  def eds_links
+    @eds_links ||= EdsLinks::Processor.new(self)
+  end
+
+  class Processor < SearchWorks::Links
+    def all
+      link_fields.map do |link_field|
+        SearchWorks::Links::Link.new(
+          text: relabel_link(link_field)['label'],
+          href: link_field['url'],
+          fulltext: show_link?(link_fields, link_field),
+          sfx: sfx_link?(link_field)
+        )
+      end
+    end
+
+    private
+
+    def relabel_link(link)
+      link = link.dup
+      map = to_link_mapping(link)
+      link['label'] = map[:label] if map.present?
+      link
+    end
+
+    # Link types are prioritized as follows:
+    #
+    # 1 and 2 are preferred, and can coexist
+    # show 3 only if there's no 1 or 2
+    # show 4 only if there's no 1-3
+    def show_link?(links, link)
+      map = to_link_mapping(link)
+      return false if map.blank? # drop links that don't have a mapping
+
+      case map[:priority]
+      when 1, 2
+        true
+      when 3, 4
+        !link_within_priority?(links, map[:priority] - 1)
+      end
+    end
+
+    # does this link have a priority <= given priority?
+    def link_within_priority?(links, priority)
+      links.each do |link|
+        map = to_link_mapping(link)
+        return true if map.present? && map[:priority] <= priority
+      end
+      false
+    end
+
+    def sfx_link?(link)
+      (to_link_mapping(link) || {})[:priority] == 3
```

Comment:
I suppose we could either check the priority or our updated label.  Tests will fail if we change priority (which will at least let us know that we needed to update this method as well).

I'm on the fence.